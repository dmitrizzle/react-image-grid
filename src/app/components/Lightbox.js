import { Button } from "@roast-cms/react-button-beans";
import React from "react";
import styled from "styled-components";

import { Title } from "./Header";
import Overlay from "./Overlay";

export const LightboxOverlay = styled(Overlay)`
  position: fixed;
  display: block;
  z-index: 2;
  cursor: pointer;
  overflow-y: scroll;
`;
export const Box = styled.div`
  max-width: 520px;
  width: 90%;
  margin: 2em auto;
  background: #fff;
  color: #000;
  border-radius: .5em;
  overflow: hidden;
  & h1 {
    padding: .5em;
    background: #eee;
  }
  & p {
    padding: 1em;
    margin: 0;
    font-family: ${props => props.theme.typography.text.font};
  }
`;
export const LightboxButton = styled(Button)`
  border-radius: 0;
  margin-bottom: 0;
  max-width: 100%;
`;
export const ImageLarge = styled.img`width: 100%;`;

export default props => {
  return (
    <LightboxOverlay onClick={props.onClick}>
      <Box>
        <Title>
          Image by {props.data.user}
        </Title>
        <ImageLarge src={props.data.src} />
        <p>
          {props.data.caption}
        </p>
        {props.data.tags &&
          props.data.tags.length > 0 &&
          <p>
            <strong>Tags:</strong>{" "}
            {props.data.tags.map((tag, count) => {
              return `${tag}${count === props.data.tags.length - 1
                ? "."
                : ","} `;
            })}
          </p>}
          <p><small>Published on {props.data.published}</small></p>
        <LightboxButton
          onClick={event => {
            event.stopPropagation();
            window.open(props.data.link);
          }}
          inverse
        >
          See on Flickr
        </LightboxButton>
      </Box>
    </LightboxOverlay>
  );
};
