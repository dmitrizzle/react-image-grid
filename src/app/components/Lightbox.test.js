import "jest-styled-components";

import { Sugar } from "@roast-cms/react-sugar-styled";
import React from "react";

import { shallow } from "enzyme";

import Lightbox, {
  LightboxOverlay,
  Box,
  LightboxButton,
  ImageLarge
} from "./Lightbox";

test("Render Box without crashing, match snapshot", () => {
  const element = shallow(<Box theme={Sugar({})} />);
  expect(element).toMatchSnapshot();
});
test("Render Lightbox without crashing, match snapshot", () => {
  const element = shallow(
    <Lightbox
      data={{
        title: "user"
      }}
    />
  );
  expect(element).toMatchSnapshot();
});
test("Render LightboxOverlay without crashing, match snapshot", () => {
  const element = shallow(<LightboxOverlay />);
  expect(element).toMatchSnapshot();
});
test("Render LightboxButton without crashing, match snapshot", () => {
  const element = shallow(<LightboxButton />);
  expect(element).toMatchSnapshot();
});
test("Render ImageLarge without crashing, match snapshot", () => {
  const element = shallow(<ImageLarge />);
  expect(element).toMatchSnapshot();
});
