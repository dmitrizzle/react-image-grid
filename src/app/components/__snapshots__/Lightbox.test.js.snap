// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Box without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    theme={
      Object {
        "color": Object {
          "background": [Function],
          "brand": [Function],
          "foreground": [Function],
          "highlight": [Function],
        },
        "effects": Object {
          "borderRadius": Object {
            "med": 0.45,
            "small": [Function],
          },
        },
        "layer": Object {
          "card": 30,
          "nav": 20,
          "overlay": 40,
          "tuck": -1,
          "up": 10,
        },
        "opacity": Object {
          "half": 0.5,
          "least": 0.125,
          "most": 0.85,
        },
        "size": Object {
          "block": Object {
            "border": 4,
            "column": Object {
              "l": 900,
              "m": 700,
            },
            "padding": 1.5,
            "spacing": 0.75,
          },
          "breakpoint": Object {
            "exact": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "max": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "min": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "stops": Object {
              "max": 1600,
              "min": 320,
            },
          },
          "font": Object {
            "auto": Array [
              "
        ",
              "
		@media (min-width: ",
              "32.5625",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "32.5",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "20",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (min-width: ",
              "100.0625",
              "em) {
			",
              "font-size:	",
              [Function],
              "px;",
              "
		}
	",
              ";
      ",
            ],
            "l": 20.7,
            "m": 18,
            "make": Object {
              "larger": 2,
              "normal": 1,
              "smaller": 0.85,
              "tiny": 0.5,
            },
            "s": 16.2,
            "xs": 15.299999999999999,
          },
        },
        "typography": Object {
          "text": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "Arial, sans-serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
        ",
            ],
            "font": "Arial, sans-serif",
            "lineHeight": 1.75,
          },
          "title": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "Arial, sans-serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
          font-weight: ",
              "700",
              ";
          margin: 0;
        ",
            ],
            "font": "Arial, sans-serif",
            "lineHeight": 1.15,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-dnqmqq jLVXtG",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-dnqmqq jLVXtG",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render ImageLarge without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.img />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-gZMcBi ePwOMm",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-gZMcBi ePwOMm",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render Lightbox without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
      Object {
        "title": "user",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.h1>
          Image by 
        </styled.h1>
        <styled.img />
        <p />
        <p>
          <small>
            Published on 
          </small>
        </p>
        <Styled(Styled(Component))
          inverse={true}
          onClick={[Function]}
        >
          See on Flickr
        </Styled(Styled(Component))>
      </styled.div>,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.h1>
            Image by 
          </styled.h1>,
          <styled.img />,
          <p />,
          undefined,
          <p>
            <small>
              Published on 
            </small>
          </p>,
          <Styled(Styled(Component))
            inverse={true}
            onClick={[Function]}
          >
            See on Flickr
          </Styled(Styled(Component))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Image by ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Image by ",
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <small>
              Published on 
            </small>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Published on ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Published on ",
              undefined,
            ],
            "type": "small",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "See on Flickr",
            "inverse": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "See on Flickr",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.h1>
            Image by 
          </styled.h1>
          <styled.img />
          <p />
          <p>
            <small>
              Published on 
            </small>
          </p>
          <Styled(Styled(Component))
            inverse={true}
            onClick={[Function]}
          >
            See on Flickr
          </Styled(Styled(Component))>
        </styled.div>,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h1>
              Image by 
            </styled.h1>,
            <styled.img />,
            <p />,
            undefined,
            <p>
              <small>
                Published on 
              </small>
            </p>,
            <Styled(Styled(Component))
              inverse={true}
              onClick={[Function]}
            >
              See on Flickr
            </Styled(Styled(Component))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Image by ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Image by ",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <small>
                Published on 
              </small>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Published on ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Published on ",
                undefined,
              ],
              "type": "small",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "See on Flickr",
              "inverse": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "See on Flickr",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render LightboxButton without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Styled(Component)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "sc-iwsKbI hzEVsK",
      "innerRef": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "sc-iwsKbI hzEVsK",
        "innerRef": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render LightboxOverlay without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(styled.aside) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "sc-htoDjs kXRwWv",
      "innerRef": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "sc-htoDjs kXRwWv",
        "innerRef": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
